nss-day-cohort-46/rare-server-tragic-trolls
# Nc adding tags

# Description

When posting a new post, you can pass an array of tag Id's and it will create join table objects for you. When getting posts, it will return with an array of tag objects.

## Type of change


# Testing Instructions

create some data by posting in postman:

```
http://localhost:8088/posts
```

with this object:
```
{
        "title": "Title with tags",
        "imageUrl": "",
        "tagIds": [2,3],
        "userId": 2,
        "categoryId": 1,
        "content": "Post Text #8",
        "publicationDate": "2026-04-02T23:57:07.374Z",
        "approved": true
    }
```
*Ensure you have tag Ids with id=2 and 3. If not, create some in your tag table.*

see that an array of tag id's gets returned when performing a get method in these 3 ways:
1.
```
http://localhost:8088/posts?userId=2
```
2.
```
http://localhost:8088/posts
```
3.
```
http://localhost:8088/posts/2
```

# Checklist:
